ModulemdTranslationHelpers/Utils.py

```diff
@@ -0,0 +1,120 @@
+    module = fedora_index.get_module(mod_name)
+
+    # Retrieve all Modulemd.ModuleStreamV2 objects from module
+    streams = module.get_all_streams()
+    for stream in streams:
+        key = stream.get_stream_name()
+        if key not in final_streams.keys():
+            final_streams[key] = stream.get_NSVCA()
+        else:
+            nsvca = stream.get_NSVCA()
+            version = int(nsvca.split(':')[2])
+            if version > int(final_streams[key].split(
+                    ":")[2]):  # found a higher version for same stream
+                final_streams[key] = nsvca
+
+summary_dict = {}
+description_dict = {}
+profile_desc_default_dict = {}
+profile_desc_minimal_dict = {}
+
+for stream_name, nsvca in final_streams.items():
+    mdname = nsvca.split(":")[0]
+    version = int(nsvca.split(":")[2])
+    context = nsvca.split(":")[3]
+    arch = nsvca.split(":")[4]
+    module = fedora_index.get_module(mdname)
+    stream = module.get_stream_by_NSVCA(stream_name, version, context, arch)
+
+    # Process description
+    description = stream.get_description("C")
+    if description not in description_dict.keys():
+        description_dict[description] = [nsvca.rsplit(
+            ":", 3)[0] + ";description"]  # name:stream
+    else:
+        description_dict[description].append(
+            nsvca.rsplit(":", 3)[0] + ";description")
+
+    # Process summary
+    summary = stream.get_summary("C")
+    if summary not in summary_dict.keys():
+        summary_dict[summary] = [
+            nsvca.rsplit(
+                ":",
+                3)[0] +
+            ";summary"]  # name:stream
+    else:
+        summary_dict[summary].append(nsvca.rsplit(":", 3)[0] + ";summary")
+
+    # Process profile descriptions(sometimes NULL)
+    profile_names = stream.get_profile_names()
+    for names in profile_names:
+        profile = stream.get_profile(names)
+        profile_desc = profile.get_description("C")
+        if profile_desc:
+            if profile_desc[0] not in profile_desc_default_dict.keys():
+                profile_desc_default_dict[
+                    profile_desc[0]] = [
+                    nsvca.rsplit(
+                        ":",
+                        3)[0] +
+                    ";profile;default"]
+            else:
+                profile_desc_default_dict[profile_desc[0]].append(
+                    nsvca.rsplit(":", 3)[0] + ";profile;default")
+
+            if profile_desc[1] not in profile_desc_minimal_dict.keys():
+                profile_desc_minimal_dict[
+                    profile_desc[1]] = [
+                    nsvca.rsplit(
+                        ":",
+                        3)[0] +
+                    ";profile;minimal"]
+            else:
+                profile_desc_minimal_dict[profile_desc[1]].append(
+                    nsvca.rsplit(":", 3)[0] + ";profile;minimal")
+
+
+catalog = Catalog(project="fedora-modularity-translations")
+
+for key, value in summary_dict.items():
```

Comment:
The net result here is that you may be adding duplicates to the catalog, which will break gettext. There should be a single dictionary that gets converted into the catalog.